backends/gdx-backends-gwt/src/com/badlogic/gwtref/client/ReflectionCache.java

```diff
@@ -19,22 +19,40 @@
 import com.google.gwt.core.client.GWT;
 
 public class ReflectionCache {
-	private static IReflectionCache instance = GWT.create(IReflectionCache.class);
+	private static IReflectionCache instance1 = GWT.create(IReflectionCache.class);
+	private static IReflectionCache instance2 = GWT.create(IReflectionCache2.class);
 
 	public static Type forName (String name) throws ClassNotFoundException {

```

Comment:
Has a throws clause for `ClassNotFoundException` but will throw a `RuntimeException` on class not found. Perhaps remove the clause?

Reply:
Good catch, this should probably use ClassNotFoundException for when a Class is not found.

Reply:
Removing ClassNotFoundException from the signature would probably force ClassReflection.forName to catch and throw runtime exceptions as Reflection exceptions, which seems more awkward. Throwing a ClassNotFoundException would make it inconsistent with getType, which can't be easily refactored without changing the public-facing API.

Reply:
Great, if this matches the public API then it should need no changes. I'm happy if this doesn't change anything for users.

Reply:
Ok I'll skip the PR. Most reflection issues fail fast and on in libGDX/GWT context typically are developer errors, so any exception works.

Just documenting for GWT users who stumble upon this ticket: GWT `ClassReflection.forName` throws a RuntimeException, on desktop it throws ReflectionException.